SliceTracker/SliceTracker.py

```diff
@@ -3576,3 +3643,196 @@ def onCaseNumberChanged(self, caseNumber):
+
+    plotFrameLayout.addWidget(self.plottingFrameWidget)
+
+    self.popupChartButton.connect('toggled(bool)', self.onDockChartViewToggled)
+    self.showLegendCheckBox.connect('stateChanged(int)', self.onShowLegendChanged)
+
+    self.addPlotPoints([[0, 0, 0], [0, 0, 0]])
+
+  def addPlotPoints(self, triplets):
+    numCurrentRows = self.chartTable.GetNumberOfRows()
+    self.chartView.removeAllPlots()
+    for i in range(0, len(triplets)):
+      if numCurrentRows == 0:
+        self._chartView.hide()
+        self.arrX.InsertNextValue(0)
+      else:
+        self._chartView.show()
+        self.arrX.InsertNextValue(numCurrentRows + i - 1)
+      self.arrXD.InsertNextValue(triplets[i][0])
+      self.arrYD.InsertNextValue(triplets[i][1])
+      self.arrZD.InsertNextValue(triplets[i][2])
+      distance = (triplets[i][0] ** 2 + triplets[i][1] ** 2 + triplets[i][2] ** 2) ** 0.5
+      self.arrD.InsertNextValue(distance)
+
+    xvals = vtk.vtkDoubleArray()
+    xlabels = vtk.vtkStringArray()
+    maxX = self.arrX.GetNumberOfValues() - 2
+    for j in range(0, maxX + 1):
+      xvals.InsertNextValue(j)
+      xlabels.InsertNextValue(str(j))
+    self.xAxis.SetCustomTickPositions(xvals, xlabels)
+    self.xAxis.SetBehavior(vtk.vtkAxis.FIXED)
+    self.xAxis.SetRange(0, maxX + 0.1)
+
+    plot = self.chart.AddPlot(vtk.vtkChart.LINE)
+    plot.SetInputData(self.chartTable, 0, 1)
+    plot.SetColor(255, 0, 0, 255)
+    vtk.vtkPlotLine.SafeDownCast(plot).SetMarkerStyle(4)
+    vtk.vtkPlotLine.SafeDownCast(plot).SetMarkerSize(3 * plot.GetPen().GetWidth())
+
+    plot2 = self.chart.AddPlot(vtk.vtkChart.LINE)
+    plot2.SetInputData(self.chartTable, 0, 2)
+    plot2.SetColor(0, 255, 0, 255)
+    vtk.vtkPlotLine.SafeDownCast(plot2).SetMarkerStyle(4)
+    vtk.vtkPlotLine.SafeDownCast(plot2).SetMarkerSize(3 * plot2.GetPen().GetWidth())
+
+    plot3 = self.chart.AddPlot(vtk.vtkChart.LINE)
+    plot3.SetInputData(self.chartTable, 0, 3)
+    plot3.SetColor(0, 0, 255, 255)
+    vtk.vtkPlotLine.SafeDownCast(plot3).SetMarkerStyle(4)
+    vtk.vtkPlotLine.SafeDownCast(plot3).SetMarkerSize(3 * plot3.GetPen().GetWidth())
+
+    plot4 = self.chart.AddPlot(vtk.vtkChart.LINE)
+    plot4.SetInputData(self.chartTable, 0, 4)
+    plot4.SetColor(0, 0, 0, 255)
+    vtk.vtkPlotLine.SafeDownCast(plot4).SetMarkerStyle(4)
+    vtk.vtkPlotLine.SafeDownCast(plot4).SetMarkerSize(3 * plot4.GetPen().GetWidth())
+
+  def resetAndInitializeChart(self):
+    self.arrX.Initialize()
+    self.arrXD.Initialize()
+    self.arrYD.Initialize()
+    self.arrZD.Initialize()
+    self.arrD.Initialize()
+    self.addPlotPoints([[0, 0, 0], [0, 0, 0]])
+    if self.showLegendCheckBox.isChecked():
+      self.showLegendCheckBox.setChecked(False)
+
+  def onShowLegendChanged(self, checked):
+    if checked == 2:
+      self.chart.SetShowLegend(True)
+      self.chartView.scene().SetDirty(True)
+    else:
+      self.chart.SetShowLegend(False)
+      self.chartView.scene().SetDirty(True)
```

Comment:
Again, thanks for this suggestion; I have gotten some shortcut/better code writing ideas from the existing code but things like this really help!